src/sqlfluff/dialects/postgres_keywords.py

```diff
@@ -0,0 +1,899 @@
+"""Keywords in the Postgres Dialect.
+
+Most of the keywords come from https://www.postgresql.org/docs/13/sql-keywords-appendix.html
+Here, "not-keyword" refers to a word not being a keyword, and will be removed from any default keyword definition,
+these keywords are, or have been, an ANSI keyword.
+
+There are also some keywords that are(n't) supported as types and function, but htere isn't support for that
+distinction at present.
+"""
+
+
+def priority_keyword_merge(*args):
+    """Merge keyword lists, giving priority to entries in later lists.
+
+    *args is a list of keyword lists, these lists should be of tuples in the form (keyword, type)
+
+    """
+    keyword_lists = [*args]
+    base_list = []
+    if len(keyword_lists) == 1:
+        return keyword_lists[0]
+
+    while len(keyword_lists) > 1:
+        base_list, priority_list = keyword_lists[0], keyword_lists[1]
+        keyword_set = set([x[0] for x in base_list])
+        for item in priority_list:
+            if item[0] in keyword_set:
+                for index, keyword in enumerate(base_list):
+                    if keyword[0] == item[0]:
+                        base_list.pop(index)
+                        break
+            base_list.append(item)
+
+        keyword_lists.pop(1)
+
+    return base_list
+
+
+def get_keywords(keyword_list, keyword_type):
+    """Get a list of keywords of the required type.
+
+    keyword_type should be one of "not-keyword", "reserved", "non-reserved"
+    """
+    keywords = [x[0] for x in keyword_list if x[1].startswith(keyword_type)]
+
+    return keywords
+
+
+postgres_docs_keywords = [
+    ("A", "not-keyword"),
+    ("ABORT", "non-reserved"),
+    ("ABS", "not-keyword"),
+    ("ABSENT", "not-keyword"),
```

Comment:
Any reason you didn't got the `ansi_keywords.py` and `exasol_keywords.py` route?

Reply:
Yeah, a few. Firstly, there are types in the list that are like "reserved-(can-be-function-or-type)", this allows us to implement this extra functionality later. Second, I wanted to track documented keywords, and non-documented separately, and when I do the PostGIS code, I want those keywords separately tracked as well. Thirdly, this includes words to deallocate as "not-keyword". And finally, the docs keywords conforms to the table in https://www.postgresql.org/docs/13/sql-keywords-appendix.html, it's harder to track and more work to pull them out into separate reserved and non-reserved lists.

Reply:
* i didn’t mean this to sound sassy or dismissive... but reason it back, it might do 😬

Reply:
No, no, it's all good. Good to hear the reasons. I'm still not 100% convinced though. But just noticed that exasol did a different implementation so maybe it doesn't need to be standard?

On the last point I do like matching the spec as much as possible (as you've noticed when I've ask you to align grammer more closely to the spec) but not as convinced on the keywords. They could change that table at any time.

Anyway not a complete deal breaker so will give it some more thought and come back to it once we resolve my first (bigger) concern above.

But good to know there was a good reason for it (several in fact!).

Reply:
They could, and I might implement an explicit test set on these keywords, to make sure we don’t lose any. I look forward to your snowflake pull... not against working on snowflake myself, I’m only working on Postgres because, I’ve worked in it enough to know the syntax well, and I noticed that you guys were looking for people to work on it